Modules/Core/Common/include/itkShapedNeighborhoodIterator.hxx

```diff
@@ -1,31 +0,0 @@
-/*=========================================================================
- *
- *  Copyright NumFOCUS
- *
- *  Licensed under the Apache License, Version 2.0 (the "License");
- *  you may not use this file except in compliance with the License.
- *  You may obtain a copy of the License at
- *
- *         https://www.apache.org/licenses/LICENSE-2.0.txt
- *
- *  Unless required by applicable law or agreed to in writing, software
- *  distributed under the License is distributed on an "AS IS" BASIS,
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *  See the License for the specific language governing permissions and
- *  limitations under the License.
- *
- *=========================================================================*/
-#ifndef itkShapedNeighborhoodIterator_hxx
-#define itkShapedNeighborhoodIterator_hxx
-namespace itk
-{
-template <typename TImage, typename TBoundaryCondition>
-void
-ShapedNeighborhoodIterator<TImage, TBoundaryCondition>::PrintSelf(std::ostream & os, Indent indent) const
-{
-  os << indent << "ShapedNeighborhoodIterator = " << this << std::endl;
```

Comment:
What will be printed now as the class name? `ConstShapedNeighborhoodIterator` or `ShapedNeighborhoodIterator`?

Reply:
Ah, haven't thought about it;  my bad. I was looking at other things when I realized about these classes having no ivar members, and remembered about this PR:
https://github.com/InsightSoftwareConsortium/ITK/pull/2715

However, it looks like all the underlying toolchain seem to make the trick for the above class:
https://github.com/InsightSoftwareConsortium/ITK/blob/23da3dfa79152648f3d38050ee147df47f41ce0a/Modules/Core/Common/test/itkBSplineInterpolationWeightFunctionTest.cxx
https://open.cdash.org/test/921810891 (see the end of the first line; the test prints other stuff without breaking the line)

vs. a class that re-implements`PrintSelf` e.g.
https://open.cdash.org/test/921810890

So it looks like the information gets printed (have not investigated the underlying mechanisms/have not checked the result of the test).